docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,510 @@
+                'provinces' => ['xpath' => '//*[contains(@class, "provinces")]'] // Now your value is an array where key is your locator type.
+            ]);
+        }
+
+        // Form now you can easy manipulate with your page elements.
+        public function addProvince(ProvinceInterface $province)
+        {
+            $provinceSelectBox = $this->getElement('provinces');
+
+            $provinceSelectBox->selectOption($province->getName());
+        }
+    }
+
+Let's back to our main example and analyse our scenario.
+We got steps like "When I choose "France" and I add it Then I should be notified that it has been successfully created And the country "France" should appear in the store".
+
+.. code-block:: php
+    namespace Sylius\Behat\Page\Admin\Country;
+
+    use Sylius\Behat\Page\Admin\Crud\CreatePage as BaseCreatePage;
+
+    class CreatePage extends BaseCreatePage implements CreatePageInterface
+    {
+        /**
+         * @param string $name
+         */
+        public function chooseName($name)
+        {
+            $this->getDocument()->selectFieldOption('Name', $name);
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function create()
+        {
+            $this->getDocument()->pressButton('Create');
+        }
+    }
+
+.. code-block:: php
+    namespace Sylius\Behat\Page\Admin\Country;
+
+    use Sylius\Behat\Page\Admin\Crud\IndexPage as BaseIndexPage;
+
+    class IndexPage extends BaseIndexPage implements IndexPageInterface
+    {
+        /**
+         * {@inheritdoc}
+         */
+        public function isSingleResourceOnPage(array $parameters)
+        {
+            try {
+                $rows = $this->tableAccessor->getRowsWithFields($this->getElement('table'), $parameters);
+                // Table accessor is a helper service which is responsible for all html table operations
+
+                return 1 === count($rows);
+            } catch (\InvalidArgumentException $exception) {
+                return false;
+            } catch (ElementNotFoundException $exception) {
+                return false;
+            }
+        }
+    }
+
+.. warning::
+
+    There is one small gap in this concept PageObjects aren't the concrete instance of current opened page they only mimic this behaviour (dummy pages).
+    This gap will be more understandable on code example.
+
+.. code-block:: php
+
+    // Of course this is only to illustrate this gap.
+
+    final class GapContext implements Context
+    {
+        private $homePage;
+        private $leagueIndexPage;
+
+        /**
+         * @Given I want to be on home page
+         */
+        public function iWantToBeOnHomePage() // After this method call we will be on "http://your_domain.com".
+        {
+            $this->homePage->open(); //When we add @javascript tag we can actually see this thanks to selenium.
+        }
+
+        /**
+         * @Then I want to read side bar and get The weather in France information
```

Comment:
`Then I want to see the sidebar and get information about the weather in France`